src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,72 @@
 'use strict';
 
+const NAME = '';
+const HEALTH = 100;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = NAME, health = HEALTH) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
+const HIDDEN = false;
```

Comment:
The constant `HIDDEN` is unnecessary since `false` can be directly used as the default value for the `hidden` property in the `Herbivore` class constructor.